pkg/apis/controller/v1alpha1/bluegreendeployment_types.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BlueGreenDeploymentSpec defines the desired state of BlueGreenDeployment.
+type BlueGreenDeploymentSpec struct {
+	// Replicas is the number of desired replicas. It determines number of pods for both
+	// Blue and Green ReplicaSets. It defaults to 1 if unspecified.
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
```

Comment:
Lets set a minimum with crd validation

Reply:
oops, i didn't add the minimum value of 1.